components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/WebsocketHandler.java

```diff
@@ -0,0 +1,437 @@
+		String resourceLevelThrottleKey = apiLevelThrottleKey;
+		String subscriptionLevelThrottleKey = appId + ":" + apiContext + ":" + apiVersion;
+		String messageId = UIDGenerator.generateURNString();
+		String remoteIP = ctx.channel().remoteAddress().toString();
+		remoteIP = remoteIP.substring(1, remoteIP.indexOf(":"));
+		JSONObject jsonObMap = new JSONObject();
+		if (remoteIP != null && remoteIP.length() > 0) {
+			jsonObMap.put(APIThrottleConstants.IP, APIUtil.ipToLong(remoteIP));
+		}
+		boolean isTrottled = util.service(resourceLevelThrottleKey, subscriptionLevelThrottleKey,
+		                                  applicationLevelThrottleKey);
+		if (isTrottled) {
+			ctx.writeAndFlush(new TextWebSocketFrame("Websocket frame throttled out"));
+			return false;
+		}
+		PrivilegedCarbonContext.endTenantFlow();
+		Object[] objects =
+				new Object[] { messageId, applicationLevelThrottleKey, applicationLevelTier,
+				               apiLevelThrottleKey, apiLevelTier, subscriptionLevelThrottleKey,
+				               subscriptionLevelTier, resourceLevelThrottleKey, resourceLevelTier,
+				               authorizedUser, apiContext, apiVersion, appTenant, apiTenant, appId,
+				               apiName, jsonObMap.toString() };
+		org.wso2.carbon.databridge.commons.Event event =
+				new org.wso2.carbon.databridge.commons.Event(
+						"org.wso2.throttle.request.stream:1.0.0", System.currentTimeMillis(), null,
+						null, objects);
+		throttleDataPublisher.getDataPublisher().tryPublish(event);
+		return true;
+	}
+
+	//---------------------------------Subscription----------------------------------------------
+
+	public boolean validateSubscriptionDetails(String context, String version, String consumerKey)
+			throws APIManagementException {
+		boolean defaultVersionInvoked = false;
+		String apiTenantDomain = MultitenantUtils.getTenantDomainFromRequestURL(context);
+		if (apiTenantDomain == null) {
+			apiTenantDomain = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+		}
+		int apiOwnerTenantId = APIUtil.getTenantIdFromTenantDomain(apiTenantDomain);
+		//Check if the api version has been prefixed with _default_
+		if (version != null && version.startsWith(APIConstants.DEFAULT_VERSION_PREFIX)) {
+			defaultVersionInvoked = true;
+			//Remove the prefix from the version.
+			version = version.split(APIConstants.DEFAULT_VERSION_PREFIX)[1];
+		}
+		String sql;
+		boolean isAdvancedThrottleEnabled = APIUtil.isAdvanceThrottlingEnabled();
+		if (!isAdvancedThrottleEnabled) {
+			if (defaultVersionInvoked) {
+				sql = SQLConstants.VALIDATE_SUBSCRIPTION_KEY_DEFAULT_SQL;
+			} else {
+				sql = SQLConstants.VALIDATE_SUBSCRIPTION_KEY_VERSION_SQL;
+			}
+		} else {
+			if (defaultVersionInvoked) {
+				sql = SQLConstants.ADVANCED_VALIDATE_SUBSCRIPTION_KEY_DEFAULT_SQL;
+			} else {
+				sql = SQLConstants.ADVANCED_VALIDATE_SUBSCRIPTION_KEY_VERSION_SQL;
+			}
+		}
+
+		Connection conn = null;
+		PreparedStatement ps = null;
+		ResultSet rs = null;
```

Comment:
rs, ps variable names doesn't make any sense. better to use meaningful variable name even though they are bit longer.